### ISSUE TITLE
Orders are able to go through without Shipping Payment.

### ISSUE THREAD

**Issue by jackiejagger** [woocommerce/woocommerce-gutenberg-products-block#3279]
## Describe the bug
Few orders are able to successfully go through without successfully charging the customer shipping. 

## To reproduce
**Typical checkout process, bug happens on random**

## Expected behavior
Checkout button should be grayed out until a shipping method is chosen. 

### WordPress (please complete the following information):
* Core version:  4.5.2
* WooCommerce version: 5.5.1
* WooCommerce Blocks version: 3.60
* Site language: English 
* Any other plugins installed:

### Desktop (please complete the following information):
* OS: All
* Browse: All

Happens across devices

**Comment by nerrad** [comment]
Hi there @jackiejagger,

Thanks for taking the time to report this. Unfortunately there's too little information here for us to reproduce this (which I've been unsuccessful in doing). Can you share more details about how the store is configured and setup where this is happening including:

- How is shipping configured?
- What Woo extensions do you have installed and active.
- What theme are you using.

Thanks!

**Comment by jackiejagger** [comment]
Hey @nerrad 

Thanks for your reply. Happy to provide additional information. 

### Shipping Configuration
Ship & Sell to Only Canada and US
Two Shipping methods : Canada Post and DHL Express - both pull live rates from the providers.
Hide Shipping Cost Till Address is Entered: True

### Active Plugins (18) 

Automated DHL Express Shipping: by HITStacks
WooCommerce Canada Post Shipping Plugin with Print Label: by PluginHive
CF7 to Spreadsheet: by Brainstorm Force
Contact Form 7: by Takayuki Miyoshi
Google Tag Manager for Wordpress: by Thomas Geiger
Facebook for WooCommerce: by Facebook
Avada Builder: by ThemeFusion
Avada Core: by ThemeFusion
Omnisend for Woocommerce: by Omnisend
Route App: by Route
Rank Math SEO: by Rank Math
W3 Total Cache: by BoldGrid
WooCommerce Blocks: by Automattic
WooCommerce Stripe Gateway: by WooCommerce
WooCommerce Multi Currency Premium: by VillaTheme
WooCommerce - ShipStation Integration: by WooCommerce
WooCommerce: by Automattic
Smush: by WPMU DEV

###Theme
Avada.

Let me know if you need any other information, happy to help!

J

**Comment by nerrad** [comment]
Thanks for the extra details.

**Comment by nerrad** [comment]
Moving this into the Porters requested pipeline. Whoever picks this up, please see if you can still reproduce this given the work we've done on improving various shipping related and order processing logic that _might_ have fixed this. If it's still an issue, please assign high priority and let me know! Otherwise it can be closed.

**Comment by budzanowski** [comment]
I was not able to reproduce the problem, but honestly, I have not tested with the exact site setup as for example I am still waiting for activation of my DHL XML test credentials. I have tried with other shipping plugins that offer live rates ( I suspected that this could be the culprit of the problem ). I have tried a couple of testing setups but I was not able to 

I am closing this for now - probably as @nerrad suggests the latest work in the shipping logic could have fixed this.

@jackiejagger ping us if the problem still exists. In that case, I suspect that this is a compatibility issue between our Gutenberg block and one of the plugins on your list. Without knowing which plugin has issues with our solution it will be really hard to find the problem. If the problem still occurs I would like to ask you to do a compatibility test using  **[How to Test for Plugin and Theme Conflicts](https://docs.woocommerce.com/document/how-to-test-for-conflicts/)**. Thank you again for highlighting this.
